cli/export.go

```diff
@@ -117,14 +117,88 @@ func runExport(args *docopt.Args, client controller.Client) error {
 		}
 	}
 
-	var artifact *ct.Artifact
-	if artifactID := release.ImageArtifactID(); artifactID != "" {
-		artifact, err = client.GetArtifact(artifactID)
+	blobstoreRelease, err := client.GetAppRelease("blobstore")
+	if err != nil {
+		return fmt.Errorf("error getting blobstore release: %s", err)
+	}
+	download := func(name, url string, release *ct.Release) error {
+		reqR, reqW := io.Pipe()
+		config := runConfig{
+			App:        mustApp(),
+			Release:    release.ID,
+			Artifacts:  release.ArtifactIDs[:1],
+			DisableLog: true,
+			Args:       []string{"curl", "--include", "--location", "--raw", url},
+			Stdout:     reqW,
+			Stderr:     ioutil.Discard,
+		}
+		if bar != nil {
+			config.Stdout = io.MultiWriter(config.Stdout, bar)
+		}
+		go func() {
+			if err := runJob(client, config); err != nil {
+				shutdown.Fatalf("error downloading %s: %s", name, err)
+			}
+		}()
+		req := bufio.NewReader(reqR)
+		var res *http.Response
+		maxRedirects := 5
+		for i := 0; i < maxRedirects; i++ {
+			res, err = http.ReadResponse(req, nil)
+			if err != nil {
+				return fmt.Errorf("error reading HTTP response: %s", err)
+			}
+			if res.StatusCode != http.StatusFound {
+				break
+			}
+		}
```

Comment:
Whilst we are reading multiple responses, they are all coming over the same TCP connection (i.e. from `curl`), so closing the response would actually break reading the body later on.